pkg/controller/secret/secret_ssh.go

```diff
@@ -116,11 +111,34 @@ func privateKeyFromPEM(pemKey []byte) (*rsa.PrivateKey, error) {
 	return privateKey, nil
 }
 
-func sshPublicKeyForPrivateKey(privateKey *rsa.PrivateKey) ([]byte, error) {
+func SSHPublicKeyForPrivateKey(privateKey *rsa.PrivateKey) ([]byte, error) {
 	publicKey, err := ssh.NewPublicKey(&privateKey.PublicKey)
 	if err != nil {
 		return nil, err
 	}
 
 	return ssh.MarshalAuthorizedKey(publicKey), nil
 }
+
+// CheckAndRegenPublicKey checks if the specified public key has length > 0 and regenerates it from the given private key
+// otherwise. The result is written into data
+func CheckAndRegenPublicKey(data map[string][]byte, publicKey, privateKey []byte) error {
+	if len(publicKey) == 0 {
+		// restore public key if private key exists
+		rsaKey, err := PrivateKeyFromPEM(privateKey)
+		if err != nil {
+			return err
+		}
+
+		publicKey, err = SSHPublicKeyForPrivateKey(rsaKey)
+		if err != nil {
+			return err
+		}
+
+		data[SecretFieldPublicKey] = publicKey
+
+		return nil
+	}
+
+	return nil
```

Comment:
Typically a matter of style and preference, but an early exit would reduce nesting a bit, in this case:

```suggestion
	if len(publicKey) > 0 {
		return nil
	}
	
	// restore public key if private key exists
	rsaKey, err := PrivateKeyFromPEM(privateKey)
	if err != nil {
		return err
	}

	publicKey, err = SSHPublicKeyForPrivateKey(rsaKey)
	if err != nil {
		return err
	}

	data[SecretFieldPublicKey] = publicKey

	return nil
```